src/linter/ui5Types/BestPractices.ts

```diff
@@ -0,0 +1,345 @@
+
+		hasManifestDefinition = !!(componentManifest &&
+		ts.isPropertyAssignment(componentManifest) &&
+		componentManifest.initializer.getText() === "\"json\"");
+	}
+
+	return {
+		routingAsyncFlag,
+		rootViewAsyncFlag,
+		hasAsyncInterface,
+		hasManifestDefinition,
+	};
+}
+
+function extractPropsRecursive(node: ts.ObjectLiteralExpression) {
+	const properties = Object.create(null) as propsRecord;
+
+	node.properties?.forEach((prop) => {
+		if (!ts.isPropertyAssignment(prop) || !prop.name) {
+			return;
+		}
+
+		const key = prop.name.getText();
+		if (prop.initializer.kind === ts.SyntaxKind.TrueKeyword) {
+			properties[key] = {value: true, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.FalseKeyword) {
+			properties[key] = {value: false, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.NullKeyword) {
+			properties[key] = {value: null, node: prop.initializer};
+		} else if (ts.isObjectLiteralExpression(prop.initializer) && prop.initializer.properties) {
+			properties[key] = {value: extractPropsRecursive(prop.initializer), node: prop.initializer};
+		} else if (ts.isArrayLiteralExpression(prop.initializer)) {
+			const resolvedValue = prop.initializer.elements.map((elem) => {
+				if (!ts.isObjectLiteralExpression(elem)) {
+					return;
+				}
+				return extractPropsRecursive(elem);
+			}).filter(($) => $) as propsRecordValueType[];
+
+			properties[key] = {value: resolvedValue, node: prop.initializer};
+		} else if (
+			(ts.isIdentifier(prop.initializer) ||
+			ts.isNumericLiteral(prop.initializer) ||
+			ts.isStringLiteral(prop.initializer)) &&
+
+			prop.initializer.getText()) {
+			properties[key] = {value: prop.initializer.getText(), node: prop.initializer};
+		} else {
+			// throw new Error("Unhandled property assignment");
+		}
+	});
+	return properties;
+}
+
+function reportResults(
+	analysisResult: AsyncInterfaceFindType,
+	reporter: SourceFileReporter,
+	classDesc: ts.ClassDeclaration
+) {
+	const {hasAsyncInterface, routingAsyncFlag, rootViewAsyncFlag, hasManifestDefinition} = analysisResult;
+
+	if (!hasManifestDefinition) {
+		reporter.addMessage({
+			node: classDesc,
+			severity: LintMessageSeverity.Warning,
+			ruleId: "ui5-linter-add-manifest", // TODO: Add rule id
+			message: "Include a manifest section into the Component.js",
+			messageDetails: "manifest.json will be loaded and used, " +
+			"but is not defined in Component's metadata section",
+		});
+	}
+
+	if (!hasAsyncInterface) {
+		// undefined has ambiguous meaning in that context.
+		// It could mean either implicit "true" or "false".
+		// To distinguish whether it's been set from manifest's config
+		// or not set at all, we'll use null.
+		if (rootViewAsyncFlag === false || rootViewAsyncFlag === undefined ||
+			routingAsyncFlag === false || routingAsyncFlag === undefined ||
+			(hasAsyncInterface === null && rootViewAsyncFlag === null && routingAsyncFlag === null)) {
+			reporter.addMessage({
```

Comment:
Currently, this message is also produced when the interface is not defined and no sap.ui5/rootView and no sap.ui5/routing sections exist. In such a case there should not be any message, as no sync behavior is triggered.

Reply:
Fixed with https://github.com/SAP/ui5-linter/pull/73/commits/29fa1dbbcf4cb0f9ea0c837cb39a794c69560338 https://github.com/SAP/ui5-linter/pull/73/commits/a7a307e6ddc459657676945b71324771de4292b7 https://github.com/SAP/ui5-linter/pull/73/commits/79cdbe79bb49518d42a412723653ff2333ca3e21 https://github.com/SAP/ui5-linter/pull/73/commits/89f11f8c964dff363f66e68f4bddd34bd77cd52f